rllib/agents/impala/vtrace_torch_policy.py

```diff
@@ -251,51 +163,198 @@ def make_time_major(policy, seq_lens, tensor, drop_last=False):
     return res
 
 
-def stats(policy: Policy, train_batch: SampleBatch) -> Dict[str, Any]:
+class VTraceOptimizer:
+    """Optimizer function for VTrace torch policies."""
 
-    return {
-        "cur_lr": policy.cur_lr,
-        "total_loss": torch.mean(torch.stack(policy.get_tower_stats("total_loss"))),
-        "policy_loss": torch.mean(torch.stack(policy.get_tower_stats("pi_loss"))),
-        "entropy": torch.mean(torch.stack(policy.get_tower_stats("mean_entropy"))),
-        "entropy_coeff": policy.entropy_coeff,
-        "var_gnorm": global_norm(policy.model.trainable_variables()),
-        "vf_loss": torch.mean(torch.stack(policy.get_tower_stats("vf_loss"))),
-        "vf_explained_var": torch.mean(
-            torch.stack(policy.get_tower_stats("vf_explained_var"))
-        ),
-    }
+    def __init__(self):
+        pass
 
+    @override(TorchPolicyV2)
+    def optimizer(
```

Comment:
Note: We should unify this across all algos. Like why can users chose between Adam and RMSProp for IMPALA, but not for DQN or any other algo?
Doesn't have to be in this PR.

Reply:
ok, added a TODO.
this may be agent / model dependent I think?